packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/property-table.tsx

```diff
@@ -0,0 +1,321 @@
+    textHeader: palette.gray[80],
+    bgHeaderHasFocus: "transparent",
+    textBubble: palette.gray[70],
+    bgBubble: palette.gray[20],
+    accentLight: palette.gray[20],
+    bgHeaderHovered: "white",
+    cellHorizontalPadding: 22,
+    baseFontStyle: "500 14px",
+    headerFontStyle: "600 14px",
+    editorFontSize: "14px",
+  };
+
+  const getContent = useCallback(
+    ([col, row]: Item): GridCell => {
+      const property = rowData[row];
+
+      if (!property) {
+        throw new Error();
+      }
+
+      const propertyKey = indexes[col];
+
+      if (!propertyKey) {
+        throw new Error();
+      }
+
+      const value = property[propertyKey];
+
+      switch (propertyKey) {
+        case "title":
+          return {
+            kind: GridCellKind.Text,
+            data: value,
+            displayData: value,
+            readonly: true,
+            allowOverlay: false,
+          };
+
+        case "dataTypes":
+          return {
+            kind: GridCellKind.Bubble,
+            data: value,
+            allowOverlay: true,
+          };
+
+        case "value":
+          if (typeof value === "number") {
+            return {
+              kind: GridCellKind.Number,
+              data: value,
+              displayData: String(value),
+              allowOverlay: true,
+              cursor: "pointer",
+            };
+          }
+
+          if (typeof value === "boolean") {
+            return {
+              kind: GridCellKind.Boolean,
+              data: value,
+              allowOverlay: false,
+            };
+          }
+
+          // everything else renders like Text for now
+          return {
+            kind: GridCellKind.Text,
+            data: value,
+            displayData: value,
+            allowOverlay: true,
+            cursor: "pointer",
+          };
+      }
+    },
+    [rowData],
+  );
+
+  const onCellEdited = useCallback(
+    async ([col, row]: Item, newValue: EditableGridCell) => {
+      const key = indexes[col];
+      const property = rowData[row];
+
+      if (!key || !property) {
+        throw new Error();
+      }
+
+      const updatedProperties = {
+        ...entity.properties,
+        [property.propertyTypeId]: newValue.data,
+      };
+
+      /**
+       * setting state for optimistic update
+       * also storing previous entity, so we can rollback if API call fails
+       */
+      const prevEntity = { ...entity };
+      setEntity({
+        ...entity,
+        properties: updatedProperties,
+      });
+
+      try {
+        await updateEntity({
+          data: {
+            entityId: entity.entityId,
+            updatedProperties,
+          },
+        });
+      } catch (error) {
+        // rollback & give feedback to user
+        // snackbar.error(`Failed to update "${propertyTitle}"`)
+        setEntity(prevEntity);
+      }
+    },
+    [rowData, entity, setEntity, updateEntity],
+  );
+
+  const drawCell: DrawCustomCellCallback = useCallback(
```

Comment:
You currently have to click twice on a cell to edit it – will that be changed in a future PR, so that we can render an input on hover?

Reply:
Added to the todo-list, going to investigate this on next PR